compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1929,6 +1955,10 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             .into_iter()
             .map(|(branch, mut candidates)| {
                 let candidate_start = self.cfg.start_new_block();
+                add_mcdc_coverage_matched_blk(
+                    candidates.first().expect("candidates must be not empty").extra_data.span,
+                    candidate_start,
+                );
```

Comment:
You can just use the new map returned from `sort_candidates` now:
```suggestion
                if let Some(span) = branch_coverage_spans.get(branch) {
                    coverage_targets.push((span, candidate_start));
                }
```